### DISCUSSION TITLE
[BUG] In version 0.115.0 of FastAPI, the pydantic model that has declared an alias cannot correctly receive query parameters

### DISCUSSION THREAD

**Discussion by insistence** [fastapi/fastapi#12401] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import uvicorn
from typing import Literal

from fastapi import FastAPI, Query
from pydantic import BaseModel, ConfigDict, Field
from pydantic.alias_generators import to_camel

app = FastAPI()


class FilterParams(BaseModel):
    model_config = ConfigDict(alias_generator=to_camel)

    limit: int = Field(100, gt=0, le=100)
    offset: int = Field(0, ge=0)
    order_by: Literal['created_at', 'updated_at'] = 'created_at'
    tags: list[str] = []


@app.get('/items/')
async def read_items(filter_query: FilterParams = Query()):
    return filter_query


if __name__ == '__main__':
    uvicorn.run(app='app:app')
```

### Description

Running the code in the example above, I encountered an incorrect result when accessing http://127.0.0.1:8000/items/?offset=1&orderBy=updated_at in the browser, orderBy did not receive successfully.
```
{
    "limit": 100,
    "offset": 1,
    "orderBy": "created_at",
    "tags": []
}
```
The correct result should be as follows
```
{
    "limit": 100,
    "offset": 1,
    "orderBy": "updated_at",
    "tags": []
}
```

### Operating System

Windows

### FastAPI Version

0.115.0

### Pydantic Version

2.9.2

### Python Version

3.9.19

**Comment by sinisaos** [comment]
@insistence I think this is the same problem as with `Form` alias ([this discussion](https://github.com/fastapi/fastapi/discussions/12139#discussioncomment-10566577)). I think this problem lies [in this lines](https://github.com/fastapi/fastapi/blob/6345147c245248302e90f0eb6371f7ab7557fd1b/fastapi/dependencies/utils.py#L764-L765). I think this should be

```python
if value is not None:
    params_to_process[field.alias] = value # change field.name to field.alias
```
Can you patch a local fastapi install with that change and give it a try? With that change, everything works in my case, both with or without `alias_generator`.

**Reply by insistence** [reply]
> @insistence I think this is the same problem as with `Form` alias ([this discussion](https://github.com/fastapi/fastapi/discussions/12139#discussioncomment-10566577)). I think this problem lies [in this lines](https://github.com/fastapi/fastapi/blob/6345147c245248302e90f0eb6371f7ab7557fd1b/fastapi/dependencies/utils.py#L764-L765). I think this should be
> 
> ```python
> if value is not None:
>     params_to_process[field.alias] = value # change field.name to field.alias
> ```
> 
> Can you patch a local fastapi install with that change and give it a try? With that change, everything works in my case, both with or without `alias_generator`.

I tried modifying it locally and it was indeed effective. Are you going to make a PR for this?

**Reply by sinisaos** [reply]
@insistence I made a [PR](https://github.com/fastapi/fastapi/pull/12411) and wrote some tests. Locally all tests passed but failed in FastAPI CI for `Pydantic V1` (probably due to `ConfigDict`). I have never done a PR for FastAPI so I need to see if there is a way to skip `Pydantic V1`. 
EDIT: I was able to get the tests to run on both Pydantic `V1` and `V2` and everything passed CI except for `check-labels` tests that I don't understand. We have to wait for the FastAPI team's response about that PR.

**Reply by insistence** [reply]
👍That's great! Thank you for your contribution.

**Comment by YuriiMotov** [comment, marked as answer]
There is an issue for this. Let's track it there: https://github.com/fastapi/fastapi/issues/12402

**Reply by tiangolo** [reply]
It seems this was solved in https://github.com/fastapi/fastapi/pull/14360, available in FastAPI 0.123.3 🎉
